code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -361,20 +371,17 @@ var/list/ai_list = list()
 		if(C)
 			C.post_status("shuttle")
 
-	return
-
-/mob/living/silicon/ai/proc/ai_cancel_call()
-	set category = "AI Commands"
-
-	if(isDead())
-		to_chat(src, "You can't send the shuttle back because you are dead!")
+/mob/living/silicon/ai/proc/ai_recall_shuttle()
+	if(!ismalf(src))
+		to_chat(usr, "You send the signal to recall the shuttle but nothing happens!")
```

Comment:
This is the main reason I'm not super happy about just using the current button... this message doesn't explain to the user that they can only recall a shuttle if malfunctioning and there's no obvious place on the wiki you would look to explain why your recall button isn't working if you're new.

Instead of doubling up on one screen button, why not just make a new recall button appear above the call button if malfing? Or change the name of that screen element if malf (maybe even give it a new icon so it becomes obvious you can do it!)

Reply:
I really don't see the point of using multiple buttons, they don't overlap and there would be no situation where you can choose which one to use, it'd be always either one or the other. I'll look into making an effect so it's more obvious

Reply:
the main problem imo is that it creates confusion for a non-malf AI (which is 95% or more of the time)

Say you're a new AI and you SEE a recall button that says recall. It's unintuitive that that button doesn't work MOST of the time and then doesn't give feedback that tells you why it's not working. You could solve this by changing its name but we have a shockingly bizarre system where the *name on the button* actually affects what it does, so that can cause problems

Reply:
It's simply not possible to make the button change its name without writing spaghetti (this system does not support it because as you said, the name of the button is the "id"), and doing so would make it harder to refactor/rewrite in the future, so I won't go that route if there are alternatives.

I think a better feedback message would do it even if that means breaking immulsions: "Only malf AIs can recall the shuttle". I think it is justified in this case. What do you think?

Reply:
There might be some way to write it so it's less immersion breaking... not saying this exactly but just as an example

`Your morality core throws an error. Recalling an emergency shuttle is a symptom of a malfunctioning artificial intelligence.`

Reply:
perfect